hub/components/RealmHeader/index.tsx

```diff
@@ -60,12 +64,31 @@ interface Props extends BaseProps {
 }
 
 export function Content(props: Props) {
-  const jupiterDirectLink = useMemo(() => {
-    if (props.token?.mint) {
-      return `https://jup.ag/swap/USDC-${props.token?.mint.toString()}?inAmount=1`;
+  const { wallet } = useWallet();
+  const connection = useWalletStore((s) => s.connection);
+  const endpoint = useMemo(() => connection.endpoint, [connection]);
+
+  const mint = useMemo(() => props.token?.mint.toString(), [props.token?.mint]);
+  const initJupiter = useCallback(() => {
```

Comment:
instead of this, i would import the scripts dynamically. You could do something like this:

```ts
interface Jupiter {
  init(args: any): any;
}

function importJupiter() {
  const script = new Promise((res, rej) => {
    const existing = document.getElementById(
      'jupiter-load-script',
    ) as HTMLScriptElement | null;

    if (existing) {
      existing.onload = res;
      existing.onerror = rej;
    } else {
      const el = document.createElement('script');
      el.onload = res;
      el.onerror = rej;
      el.id = 'jupiter-load-script';
      el.type = 'text/javascript';
      el.src = 'https://terminal.jup.ag/m ain.js';
      document.head.append(el);
    }
  });

  const css = new Promise((res, rej) => {
    const existing = document.getElementById(
      'jupiter-load-styles',
    ) as HTMLLinkElement | null;

    if (existing) {
      existing.onload = res;
      existing.onerror = rej;
    } else {
      const el = document.createElement('link');
      el.onload = res;
      el.onerror = rej;
      el.id = 'jupiter-load-styles';
      el.rel = 'stylesheet';
      el.href = 'https://terminal.jup.ag/main.css';
      document.head.append(el);
    }
  });

  return Promise.all([script, css]).then(() => {
    return (window as any).Jupiter as Jupiter;
  });
}
```

then, in the click handler, you could do the following:

```ts
onClick={async () => {
  const jupiter = await importJupiter();
  jupiter.init({
    /* ... */
  }) 
}}
```

Reply:
works great!